### PR TITLE
Added types from sync-storage library for React Native

### PR DESCRIPTION
**Pull request by Dannomac**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -53,5 +53,8 @@
             "_comment": "This will remove husky from when we started migrating to use prettier",
             "pre-commit": "npm uninstall husky"
         }
+    },
+    "dependencies": {
+        "sync-storage": "^0.4.2"
```
Comment (fdebijl) [inline package.json:58]: You shouldn't touch the root `package.json`, the package itself will automatically get installed in the CI flow.

File: types/sync-storage/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+// Type definitions for sync-storage 0.4
+// Project: https://github.com/raphaelpor/sync-storage
+// Definitions by: Dannomac <https://github.com/Dannomac>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+/*
+*  Init prepares the SyncStorage to work synchronously, by getting all values for all keys previously saved on RN AsyncStorage.
+*/
+export function init(): any;
```
Comment (fdebijl) [inline types/sync-storage/index.d.ts:8]: The return type should probably be `void` here

File: types/sync-storage/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+/*
+*  Returns value of key
+*/
+export function get(key: string): any;
```
Comment (fdebijl) [inline types/sync-storage/index.d.ts:12]: Can this be more exact than `any`? I imagine local storage can really only return strings, unless this package does automatic type casting

File: types/sync-storage/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+*/
+export function set(key: string, value: any): Promise<void>;
+/*
+*  Removes value from memory and AsyncStorage. Also returns a promise.
+*/
+export function remove(key: string): Promise<void>;
+/*
+*  Returns all stored keys in an array.
+*/
+export function getAllKeys(): any[];
```
Comment (fdebijl) [inline types/sync-storage/index.d.ts:24]: This should probably also be an array of strings.

File: types/sync-storage/tslint.json [hunk 5]
```diff
@@ -0,0 +1 @@
+{ "extends": "dtslint/dt.json", "extends": "@definitelytyped/dtslint/dt.json" }
```
Comment (fdebijl) [inline types/sync-storage/tslint.json:1]: You only need the first or the second key, not both. The readme currently prescribes `{ "extends": "@definitelytyped/dtslint/dt.json" }`, so I'd go with that.

File: package.json (hunk 1 above)
Comment (Dannomac) [reply package.json]: Sorry about this. I have fixed this issue.

File: types/sync-storage/index.d.ts (hunk 2 above)
Comment (Dannomac) [reply types/sync-storage/index.d.ts]: Your right. I have updated this in my latest commit.

File: types/sync-storage/index.d.ts (hunk 3 above)
Comment (fdebijl) [reply types/sync-storage/index.d.ts:12]: That works then

**Review by fdebijl — APPROVED** [review]
Seems like a good start for typings for this package, thank you for your efforts!

**Review by jakebailey — APPROVED** [review]

**Comment by Dannomac** [comment]
Ready to merge

**Pull request merged**